controllers/appwrapper_controller.go

```diff
@@ -101,29 +90,54 @@ func (r *AppWrapperReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 		klog.Error(err, "unable to fetch appwrapper")
 	}
-
 	// todo: Move the getOCMClusterID call out of reconcile loop.
 	// Only reason we are calling it here is that the client is not able to make
 	// calls until it is started, so SetupWithManager is not working.
 	if !useMachineSets && ocmClusterID == "" {
 		getOCMClusterID(r)
 	}
 
-	stopper := make(chan struct{})
-	defer close(stopper)
-	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc:    onAdd,
-		UpdateFunc: onUpdate,
-		DeleteFunc: onDelete,
-	})
-	go informer.Run(stopper)
-	if !cache.WaitForCacheSync(stopper, informer.HasSynced) {
-		klog.Info("Wait for cache to sync")
+	//onAdd replacement
+	if appwrapper.Status.State == arbv1.AppWrapperStateEnqueued || appwrapper.Status.State == "" {
+		//scaledAppwrapper = append(scaledAppwrapper, aw.Name)
+		demandPerInstanceType := discoverInstanceTypes(&appwrapper)
+		//TODO: simplify the looping
+		if useMachineSets {
+			if r.canScaleMachineset(ctx, demandPerInstanceType) {
+				r.scaleUp(ctx, &appwrapper, demandPerInstanceType)
+			} else {
+				klog.Infof("Cannot scale up replicas max replicas allowed is %v", maxScaleNodesAllowed)
+			}
+		} else {
+			if canScaleMachinepool(demandPerInstanceType) {
+				r.scaleUp(ctx, &appwrapper, demandPerInstanceType)
+			} else {
+				klog.Infof("Cannot scale up replicas max replicas allowed is %v", maxScaleNodesAllowed)
+			}
+		}
+
 	}
-	//TODO: do we need dual sync??
-	msInformerHasSynced = informer.HasSynced()
-	addAppwrappersThatNeedScaling()
-	<-stopper
+
+	//onDelete replacement
+	if useMachineSets {
```

Comment:
That block should only be called when the AppWrapper is being deleted. We may have to rely on a finaliser, to make sure the logic is performed, even if the deletion occurs when the controller is down.